From: Lee Duncan <lduncan@suse.com>
Date: Thu 29 Sep 2022 09:46:47 AM PDT
Subject: [PATCH] kABI: fix adding another field to scsi_device
Patch-mainline: Never, kABI fix
References: bsc#1203039

Patch scsi-core-Add-BLIST_NO_ASK_VPD_SIZE-for-some-VDASD.patch
added field no_ask_vpd_sz_first, so hide that from the kABI
checker. There should be plenty of room for one more bit.

[lduncan: refreshed to apply]
---
 include/scsi/scsi_device.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/include/scsi/scsi_device.h
+++ b/include/scsi/scsi_device.h
@@ -210,10 +210,10 @@ struct scsi_device {
 	unsigned rpm_autosuspend:1;	/* Enable runtime autosuspend at device
 					 * creation time */
 #ifndef __GENKSYMS__
-	/* kABI: there should be room for this single bit */
+	/* kABI: there should be room for 2 more bits here */
 	unsigned silence_suspend:1;	/* Do not print runtime PM related messages */
-#endif
 	unsigned no_vpd_size:1;		/* No VPD size reported in header */
+#endif
 
 	unsigned int queue_stopped;	/* request queue is quiesced */
 	bool offline_already;		/* Device offline message logged */
